c10/util/TypeCast.h

```diff
@@ -48,12 +63,24 @@ template <typename dest_t, typename src_t>
 struct static_cast_with_inter_type {
   C10_HOST_DEVICE __ubsan_ignore_undefined__ static inline dest_t apply(
       src_t src) {
-    constexpr bool real = needs_real<dest_t, src_t>::value;
+    constexpr bool real = needs_real_or_bool<dest_t, src_t>::value;
     auto r = maybe_real<real, src_t>::apply(src);
     return static_cast<dest_t>(r);
   }
 };
 
+// Partial template specialization for casting to bool.
+// Need to handle complex types separately, as we don't 
+// simply want to cast the real part to bool. 
+template <typename src_t>
+struct static_cast_with_inter_type<bool, src_t> {
+  C10_HOST_DEVICE __ubsan_ignore_undefined__ static inline bool apply(
```

Comment:
I don't think we have any UB wrapper here?

Reply:
Yeah it's not necessary I think